File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {

    public int minEatingSpeed(int[] piles, int h) {

        // Descobrir maior K - criar um range entre 1 - maior K
        // Iterar por busca binária descobrindo qual o menor

        int maxK = 1;
        for (int i = 0; i < piles.length; i++) {
            if (piles[i] > maxK)
                maxK = piles[i];
        }

        int left = 1;
        int right = maxK;

        while (left < right) {
            int mid = left + (right - left) / 2;

            int temp = 0;
            for (int i =0; i < piles.length; i++) {
                temp += (piles[i] + mid - 1)/ mid;
            }

            if (temp <= h) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }

        return left;
    }
}

File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        
        resLeft = resRight = 0
        def expand(left, right):
            nonlocal resLeft, resRight
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1
                right += 1
            
            if right - left > resRight - resLeft:
                resRight = right
                resLeft = left +1

        
        for i in range(len(s)):
            expand(i, i)
            expand(i, i + 1)
        return s[resLeft:resRight]